plugins/modules/api_info.py

```diff
@@ -0,0 +1,264 @@
+      reopen_count: ""
+      reopened_by: ""
+      reopened_time: ""
+      resolved_at: "2020-09-24 19:54:48"
+      resolved_by: 6816f79cc0a8016401c5a33be04be441
+      rfc: ""
+      route_reason: ""
+      service_offering: ""
+      severity: "3"
+      short_description: EMAIL is slow when an attachment is involved
+      sla_due: ""
+      state: "7"
+      subcategory: ""
+      sys_class_name: incident
+      sys_created_by: don.goodliffe
+      sys_created_on: "2020-07-07 23:18:07"
+      sys_domain: global
+      sys_domain_path: /
+      sys_id: 46cebb88a9fe198101aee93734f9768b
+      sys_mod_count: "5"
+      sys_tags: ""
+      sys_updated_by: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
+      sys_updated_on: "2020-09-24 19:54:48"
+      task_effective_number: INC0000013
+      time_worked: ""
+      universal_request: ""
+      upon_approval: ""
+      upon_reject: ""
+      urgency: "1"
+      user_input: ""
+      watch_list: ""
+      work_end: ""
+      work_notes: ""
+      work_notes_list: ""
+      work_start: ""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, client, errors, table, utils
+from ..module_utils.api import (
+    transform_query_to_servicenow_query,
+    POSSIBLE_FILTER_PARAMETERS,
+    table_name,
+    FIELD_COLUMNS_NAME,
+)
+
+
+def run(module, table_client):
+    columns = ",".join([field.lower() for field in module.params[FIELD_COLUMNS_NAME]])
+    search_dict = dict(columns=columns)
+    search_dict.update(module.params)
+    query = utils.filter_dict(search_dict, *POSSIBLE_FILTER_PARAMETERS)
+    servicenow_query = transform_query_to_servicenow_query(query)
+    return table_client.list_records(table_name(module), servicenow_query)
+
+
+def main():
+    arg_spec = dict(
+        arguments.get_spec("instance", "sys_id"),
+        resource=dict(type="str", required=True),
+        sysparm_query=dict(type="str"),
+        display_value=dict(
+            type="str", choices=["true", "false", "both"]
+        ),  # Return field display values (true), actual values (false), or both (all) (default: false)
+        exclude_reference_link=dict(
+            type="bool",
+        ),  # True to exclude Table API links for reference columns
+        columns=dict(
+            type="list", default=[], elements="str"
+        ),  # A comma-separated list of fields to return in the response
+        query_category=dict(
+            type="str"
+        ),  # Name of the query category (read replica category) to use for queries
+        query_no_domain=dict(
+            type="bool"
+        ),  # True to access data across domains if authorized (default: false)
+        no_count=dict(
+            type="bool",
+        ),  # Do not execute a select count(*) on table (default: false)
+    )
+
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec=arg_spec,
+        mutually_exclusive=[("sys_id", "sysparm_query")],
```

Comment:
Maybe this no longer applies? We can make `sys_id` and `sysparm query` mutually exclusive, but ServiceNow does not seem to care if both `sys_id` and other query parameters are passed in a query. In fact, `sys_id` can be a part of `sysparm_query`, which we do not even check.